File: terraingen/src/waypoint.py
from geopoint import GeoPoint

class Waypoint:
    def __init__(self):
        self.location = GeoPoint()
        self.altitude = 0
        self.name = ""
        
    def __str__(self):
        return str(self.name) + ", " + str(self.location) + ", " + str(self.altitude)

File: src/Engine/Waypoint/Waypoint.cpp
// SPDX-License-Identifier: GPL-2.0-or-later
// Copyright The XCSoar Project

#include "Waypoint.hpp"
#include "Geo/Flat/FlatProjection.hpp"

Waypoint::Waypoint(const GeoPoint &_location) noexcept
  :location(_location)
{
}

bool
Waypoint::IsCloseTo(const GeoPoint &_location,
                    const double range) const noexcept
{
  return location.Distance(_location) <= range;
}

void
Waypoint::Project(const FlatProjection &projection) noexcept
{
  flat_location = projection.ProjectInteger(location);

#ifndef NDEBUG
  flat_location_initialised = true;
#endif
}
